### PR TITLE
Allow for custom options

### PR DESCRIPTION
**Pull request by g10ria**
To add a custom option: put the .vue file in components and include/export it in OpsiOptions.ts

### CODE REVIEW

**Comment by 132ikl** [comment]
For clarity, it's probably better if it's just called "OptionPlugin"/Options.ts since as long as we don't import OptionPlugin from baklava there's no conflicts.

**Review by 132ikl — APPROVED** [review]

File: src/components/Options.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import {
+    ButtonOption,
```
Comment (carbotaniuman) [inline src/components/Options.ts]: Run the linter on this please, I'm pretty sure this isn't formatted

File: src/components/OptionPlugin.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            }
+        });
+        editor.plugins.forEach((p) => {
+            if (p.type === "ViewPlugin") {
+                this.OpsiRegisterOptions(p as IViewPlugin); // modified
+            }
+        });
+    }
+
+    private OpsiRegisterOptions(viewPlugin: IViewPlugin) {
```
Comment (carbotaniuman) [inline src/components/OptionPlugin.ts]: Methods/functions start be camelCase

File: src/components/Options.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import {
```
Comment (carbotaniuman) [inline src/components/Options.ts]: .ts files that are not classes should be camelCase

**Comment by carbotaniuman** [comment]
Also I would prefer if you put all the stuff in a new folder, such as `baklava-options-plugin`.

File: src/components/baklava-options-plugin/OptionPlugin.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+/* eslint-disable */
```
Comment (carbotaniuman) [inline src/components/baklava-options-plugin/OptionPlugin.ts]: Camel case this too please. Also specify the exact rule in eslint-disable. Other than that, looks great.

**Review by carbotaniuman — CHANGES_REQUESTED** [review]

File: src/components/baklava-options-plugin/OptionPlugin.ts (hunk 4 above)
Comment (g10ria) [reply src/components/baklava-options-plugin/OptionPlugin.ts]: I thought that since it's a class it should be upper camel case?

Comment (carbotaniuman) [reply src/components/baklava-options-plugin/OptionPlugin.ts]: I was mistaken, not for file names.

**Review by carbotaniuman — APPROVED** [review]

**Pull request merged**